modules/juce_opengl/opengl/juce_OpenGLGraphicsContext.cpp

```diff
@@ -114,7 +122,20 @@ struct CachedImageList  : public ReferenceCountedObject,
             if (textureNeedsReloading && pixelData != nullptr)
             {
                 textureNeedsReloading = false;
-                texture.loadImage (Image (*pixelData));
+#if JUCE_MAC
+                Image image;
+                if (lastDisplay != nullptr)
+                {
+                    image = juce_returnImageWithNativeColourSpace (Image (*pixelData), lastDisplay);
+                }
+                else
+                {
+                    image = Image (*pixelData);
+                }
+#else
+                auto image = Image (*pixelData);
+#endif
+                texture.loadImage (std::move (image));

```

Comment:
This code would be much nicer (and similar to the original) if you refactor out a function `imageFromPixelData`
(it would also allow you to do early `return`s within that function)